packages/ember-glimmer/lib/environment.js

```diff
@@ -133,6 +137,18 @@ export default class Environment extends GlimmerEnvironment {
           wrapClassAttribute(args);
           return new CurlyComponentSyntax({ args, definition, templates });
         }
+      } else {
+        // Check if it's a keyword
+        let mappedKey = keywords[key];
+        if (mappedKey) {
+          if (mappedKey !== key) {
+            path = path.map((segment) => segment === key ? mappedKey : segment);
```

Comment:
It's going from `textarea` to `-text-area`. It's because `-text-area` is the [registered name of the textarea component](https://github.com/emberjs/ember.js/blob/master/packages/ember-application/lib/system/application.js#L1103).

htmlbars is does the same [name re-mapping](https://github.com/emberjs/ember.js/blob/master/packages/ember-htmlbars/lib/keywords/textarea.js#L193).